### ISSUE TITLE
$S-ing a -Spec type puts its toString into a string

### ISSUE THREAD

**Issue by JakeWharton** [square/javapoet#728]
I feel like this is never what you want, and that it should behave like `"$N"`. Or is this too magical and I should just be using `"$N"`?

**Comment by ronshapiro** [comment]
FWIW, I wrote an ErrorProne check to detect usages of mistaken codeblock
arguments and I think it would catch this. In my mind, $S should almost
always be a String (or maybe CharSequence).

**Comment by sormuras** [comment]
> [...] `$S` should almostalways be a String (or maybe CharSequence).

This.

**Comment by JakeWharton** [comment]
I don't agree. The implicit behavior of `toString`-ing is fine. It's like string concatenation. My issue here is more like how `toString` on array is useless and has to be special cased. `toString()` on JavaPoet types in `$S` are useless and could be special cased.
